src/main/kotlin/com/terraformation/backend/accelerator/db/SubmissionStore.kt

```diff
@@ -27,11 +31,70 @@ class SubmissionStore(
     private val dslContext: DSLContext,
     private val eventPublisher: ApplicationEventPublisher,
 ) {
-  fun fetchOneById(
-      submissionId: SubmissionId,
-  ): ExistingSubmissionModel {
-    return fetch(SUBMISSIONS.ID.eq(submissionId)).firstOrNull()
-        ?: throw SubmissionNotFoundException(submissionId)
+  fun createSubmission(
+      deliverableId: DeliverableId,
+      projectId: ProjectId,
+  ) {
+    requirePermissions { createSubmission(projectId) }
+
+    val now = clock.instant()
+    val userId = currentUser().userId
+
+    dslContext
+        .insertInto(SUBMISSIONS)
+        .set(SUBMISSIONS.CREATED_BY, userId)
+        .set(SUBMISSIONS.CREATED_TIME, now)
+        .set(SUBMISSIONS.DELIVERABLE_ID, deliverableId)
+        .set(SUBMISSIONS.MODIFIED_BY, userId)
+        .set(SUBMISSIONS.MODIFIED_TIME, now)
+        .set(SUBMISSIONS.PROJECT_ID, projectId)
+        .set(SUBMISSIONS.SUBMISSION_STATUS_ID, SubmissionStatus.NotSubmitted)
+        .onConflict()
+        .doUpdate()
+        .set(SUBMISSIONS.MODIFIED_BY, userId)
+        .set(SUBMISSIONS.MODIFIED_TIME, now)
+        .set(SUBMISSIONS.SUBMISSION_STATUS_ID, SubmissionStatus.NotSubmitted)
+        .execute()
+  }
+
+  fun fetchActiveSpeciesDeliverableSubmission(
+      projectId: ProjectId,
+  ): ExistingSpeciesDeliverableSubmissionModel {
+    requirePermissions { readProjectDeliverables(projectId) }
+
+    val today = LocalDate.ofInstant(clock.instant(), TimeZones.UTC)
+
+    val submission =
+        dslContext
+            .select(DELIVERABLES.ID, SUBMISSIONS.ID)
+            .from(DELIVERABLES)
+            .join(MODULES)
+            .on(DELIVERABLES.MODULE_ID.eq(MODULES.ID))
+            .join(COHORT_MODULES)
+            .on(MODULES.ID.eq(COHORT_MODULES.MODULE_ID))
+            .join(PARTICIPANTS)
+            .on(COHORT_MODULES.COHORT_ID.eq(PARTICIPANTS.COHORT_ID))
+            .join(PROJECTS)
+            .on(PARTICIPANTS.ID.eq(PROJECTS.PARTICIPANT_ID))
+            .fullOuterJoin(SUBMISSIONS)
+            .on(
+                DELIVERABLES.ID.eq(SUBMISSIONS.DELIVERABLE_ID),
+                PROJECTS.ID.eq(SUBMISSIONS.PROJECT_ID))
+            .join(ORGANIZATIONS)
+            .on(PROJECTS.ORGANIZATION_ID.eq(ORGANIZATIONS.ID))
+            .where(COHORT_MODULES.START_DATE.lessOrEqual(today))
+            .and(COHORT_MODULES.END_DATE.greaterOrEqual(today))
+            .and(PROJECTS.ID.eq(projectId))
+            .and(DELIVERABLES.DELIVERABLE_TYPE_ID.eq(DeliverableType.Species))
+            .orderBy(DELIVERABLES.ID, PROJECTS.ID)
```

Comment:
Project ID is a constant so won't affect ordering. But if there's more than one species list deliverable in the module, `fetchOne` will throw an exception. So no need for an `orderBy` here at all.